utils/markdown/index.test.js

```diff
@@ -169,4 +169,48 @@ this is long text this is long text this is long text this is long text this is
         const plugin = format('[plugin](/reiciendis-0/plugins/example))', {siteURL: 'http://localhost'});
         expect(plugin).toContain('<a class="theme markdown__link" href="/reiciendis-0/plugins/example" rel="noreferrer" data-link="/reiciendis-0/plugins/example">plugin</a>');
     });
+
+    describe('should correctly open links in the current tab based on whether they are handled by the web app', () => {
+        for (const testCase of [
+            {name: 'regular link', link: 'https://example.com', handled: false},
+            {name: 'www link', link: 'www.example.com', handled: false},
+
+            {name: 'link to a channel', link: 'http://localhost/team/channels/foo', handled: true},
+            {name: 'link to a DM', link: 'http://localhost/team/messages/@bar', handled: true},
+            {name: 'link to the system console', link: 'http://localhost/admin_console', handled: true},
+            {name: 'permalink', link: 'http://localhost/reiciendis-0/pl/b3hrs3brjjn7fk4kge3xmeuffc', handled: true},
+            {name: 'link to a specific system console page', link: 'http://localhost/admin_console/plugins/plugin_com.github.matterpoll.matterpoll', handled: true},
+
+            {name: 'relative link', link: '/', handled: true},
+            {name: 'relative link to a channel', link: '/reiciendis-0/channels/b3hrs3brjjn7fk4kge3xmeuffc', handled: true},
+            {name: 'relative link to a DM', link: '/reiciendis-0/messages/b3hrs3brjjn7fk4kge3xmeuffc', handled: true},
+            {name: 'relative permalink', link: '/reiciendis-0/pl/b3hrs3brjjn7fk4kge3xmeuffc', handled: true},
+            {name: 'relative link to the system console', link: '/admin_console', handled: true},
+            {name: 'relative link to a specific system console page', link: '/admin_console/plugins/plugin_com.github.matterpoll.matterpoll', handled: true},
+
+            {name: 'link to a plugin-handled path', link: 'http://localhost/plugins/example', handled: false},
+            {name: 'link to a file attachment public link', link: 'http://localhost/files/o6eujqkmjfd138ykpzmsmc131y/public?h=j5nPX8JlgUeNVMOB3dLXwyG_jlxlSw4nSgZmegXfpHw', handled: false},
+
+            {name: 'relative link to a plugin-handled path', link: '/plugins/example', handled: false},
+            {name: 'relative link to a file attachment public link', link: '/files/o6eujqkmjfd138ykpzmsmc131y/public?h=j5nPX8JlgUeNVMOB3dLXwyG_jlxlSw4nSgZmegXfpHw', handled: false},
+
+            {name: 'link to a managed resource', link: 'http://localhost/trusted/jitsi', options: {managedResourcePaths: ['trusted']}, handled: false},
+            {name: 'relative link to a managed resource', link: '/trusted/jitsi', options: {managedResourcePaths: ['trusted']}, handled: false},
+            {name: 'link that is not to a managed resource', link: 'http://localhost/trusted/jitsi', options: {managedResourcePaths: ['jitsi']}, handled: true},
```

Comment:
@dpanic To make sure we're on the same page with how this should work, do these cases look correct to you? If you specify `trusted` as a managed resource path, `http://mymattermostserver.com/trusted/jitsi` and `/trusted/jitsi` should both be opened in a new window, but `http://mymattermostserver.com/somethingelse/trusted` should not, correct?

Reply:
Yes that was expected behaviour.